mypy/types.py

```diff
@@ -1130,6 +1130,13 @@ def relevant_items(self) -> List[Type]:
         else:
             return [i for i in self.items if not isinstance(i, NoneTyp)]
 
+    def iter_deep(self) -> Iterator[Type]:
```

Comment:
Also currently unions are flattened on creation in ``__init__`` with the help of ``self.items = flatten_nested_unions(items)``